src/Chemistry.hpp

```diff
@@ -31,15 +31,21 @@ AMREX_GPU_DEVICE void chemburner(burn_t &chemstate, Real dt);
 template <typename problem_t> void computeChemistry(amrex::MultiFab &mf, const Real dt, const Real max_density_allowed, const Real min_density_allowed)
 {
 
+	// Start off by assuming a successful burn.
+	int burn_success = 1;
+
+#if defined(AMREX_USE_GPU)
+	amrex::Gpu::Buffer<int> d_num_failed({0});
+	auto *p_num_failed = d_num_failed.data();
+#endif
+
+	int num_failed = 0;
+
 	const BL_PROFILE("Chemistry::computeChemistry()");
 	for (amrex::MFIter iter(mf); iter.isValid(); ++iter) {
 		const amrex::Box &indexRange = iter.validbox();
 		auto const &state = mf.array(iter);
 
-		if (dt < 0) {
-			amrex::Abort("Cannot do chemistry with dt < 0!");
-		}
-
 		amrex::ParallelFor(indexRange, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
```

Comment:
You can try changing this line to:
`amrex::ParallelFor(indexRange, [=, &num_failed] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {`

Reply:
This is necessary because of https://en.cppreference.com/w/cpp/language/lambda#Lambda_capture.